### DISCUSSION TITLE
boost-filesystem problem

### DISCUSSION THREAD

**Discussion by elpako** [microsoft/vcpkg#33271] [Q&A] [answered]
Hi guys! :) I'm trying to use boost-filesystem on windows with cmake and vcpkg. With default x64 windows triplet in following way:

**vcpkg.json**

```
{
    "dependencies": [
        "fmt",
        "boost",
        "boost-filesystem"
    ]
}
```

**CMakeList.txt:**

```
cmake_minimum_required(VERSION 3.27.0)

project(cmake_vcpkg_cpp_test)

find_package(boost REQUIRED COMPONENTS filesystem)
find_package(fmt REQUIRED)

add_executable(
    cmake_vcpkg_cpp_test 
    main.cpp
)

target_link_libraries (cmake_vcpkg_cpp_test PRIVATE fmt::fmt Boost::filesystem)
```

**CMakePresets.json:**
```

{
    "version": 3,
    "configurePresets": [
        {
            "name": "default",
            "displayName": "default",
            "generator": "Visual Studio 17 2022",
            "binaryDir": "${sourceDir}/build/default",
            "architecture": "x64",
            "toolchainFile": "${sourceDir}/vcpkg/scripts/buildsystems/vcpkg.cmake"
            
        }
    ],
    "buildPresets": [
        {
            "name": "default",
            "configurePreset": "default"
        }
    ]
}
```

I get the following error after cmake 

> CMake Error at CMakeLists.txt:13 (target_link_libraries):
>   Target "cmake_vcpkg_cpp_test" links to:
> 
>     Boost::filesystem
> 
>   but the target was not found.  Possible reasons include:
> 
>     * There is a typo in the target name.
>     * A find_package call is missing for an IMPORTED target.
>     * An ALIAS target is missing.

It works like charm with fmt only, Im able to compile and use fmt from vcpkg.  
I've checked this on two others machines with the same result. 

Any suggestions how to use boost:: filesystem with vcpkg? What am I doing wrong?

**Comment by Osyotr** [comment, marked as answer]
`find_package(boost REQUIRED COMPONENTS filesystem)` -> `find_package(Boost REQUIRED COMPONENTS filesystem)`?

**Reply by elpako** [reply]
Oh man, not enough focus!!! Thanks a lot!
